hardware_interface/src/component_parser.cpp

```diff
@@ -629,6 +654,218 @@ std::vector<HardwareInfo> parse_control_resources_from_urdf(const std::string &
     ros2_control_it = ros2_control_it->NextSiblingElement(kROS2ControlTag);
   }
 
+  // get min/max from ros2_control tag
```

Comment:
These lambdas are way too large for my taste. Could you please convert them to member functions? At least some of them...

Reply:
Sure, I can do that

Reply:
Done

Reply:
thank you!